tests/lib/rules/no-undef.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Tests for rule "no-undef"
+ * @author Pig Fang <g-plane@hotmail.com>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-undef");
+const RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("no-undef", rule, {
+    valid: [
+        "var test"
+    ],
+    invalid: [
+        {
+            code: "var test; text",
+            output: "var test; test",
+            errors: [{ messageId: "undef", type: "Identifier" }]
+        },
+        {
```

Comment:
what happens if the code is 
```js
var bat;
cat;
hat;
```

Reply:
It depends on the behavior of the library "string-similarity".